src/main/python/run_trecqa.py

```diff
@@ -6,24 +6,35 @@
 from pyserini import Pyserini
 from sm_cnn.bridge import SMModelBridge
 
-
-def get_answers(pyserini, question, num_hits, k, model_choice, index_path):
+class QAModel:
+  instance = None
+  def __init__(self, model_choice, index_path, qa_model_file=None):
+    if not QAModel.instance:
+       path = os.getcwd() + '/..'
+       if model_choice == "sm":
+         model_file = path + '/models/sm_model/sm_model.fixed_ext_feats_paper.puncts_stay'
+         if qa_model_file:
+           model_file = qa_model_file
+         QAModel.instance = SMModelBridge(model_file, 
+                             path + '/data/word2vec/aquaint+wiki.txt.gz.ndim=50.cache',
+                             index_path)
+
+
+def get_answers(pyserini, question, num_hits, k, model_choice, index_path, qa_model_file=None):
   candidate_passages_scores = pyserini.ranked_passages(question, num_hits, k)
   idf_json = pyserini.get_term_idf_json()
   candidate_passages = []
   tokeninzed_answer = []
 
   if model_choice == 'sm':
-    path = os.getcwd() + '/..'
 
-    model = SMModelBridge( path +'/models/sm_model/sm_model.fixed_ext_feats_paper.puncts_stay',
-                          path + '/data/word2vec/aquaint+wiki.txt.gz.ndim=50.cache',index_path)
+    qa_DL_model = QAModel(model_choice, index_path, qa_model_file).instance
```

Comment:
I'd like to think that "DL" enforces that it represents an acronym (deep learning). For instance we could have qa_IDF_model, qa_BM25_model, qa_PRM_model (positional relevance)....
Do let me know your views.